src/hooks/useKnockoutExplorer.ts

```diff
@@ -69,16 +70,38 @@ export function useKnockoutExplorer(platform: Platform): Explorer {
 
 async function configureHosted(): Promise<Explorer> {
   const win = (window as unknown) as HostedExplorerChildFrame;
+  let explorer: Explorer;
   if (win.hostedConfig.authType === AuthType.EncryptedToken) {
-    return configureHostedWithEncryptedToken(win.hostedConfig);
+    explorer = configureHostedWithEncryptedToken(win.hostedConfig);
   } else if (win.hostedConfig.authType === AuthType.ResourceToken) {
-    return configureHostedWithResourceToken(win.hostedConfig);
+    explorer = configureHostedWithResourceToken(win.hostedConfig);
   } else if (win.hostedConfig.authType === AuthType.ConnectionString) {
-    return configureHostedWithConnectionString(win.hostedConfig);
+    explorer = configureHostedWithConnectionString(win.hostedConfig);
   } else if (win.hostedConfig.authType === AuthType.AAD) {
-    return configureHostedWithAAD(win.hostedConfig);
+    explorer = await configureHostedWithAAD(win.hostedConfig);
+  } else {
+    throw new Error(`Unknown hosted config: ${win.hostedConfig}`);
   }
-  throw new Error(`Unknown hosted config: ${win.hostedConfig}`);
+
+  window.addEventListener(
```

Comment:
See if there's an easy to plug a `removeEventListener()` to some `dispose()` function for cleanup. But I wouldn't spend too much time on this, since we normally don't worry about cleaning up explorer itself.